picard/script/functions.py

```diff
@@ -681,11 +681,14 @@ def func_ne(parser, x, y):
 
 
 @script_function(documentation=N_(
-    """`$lt(x,y)`
+    """`$lt(x,y[,text])`
 
-Returns true if `x` is less than `y`."""
+Returns true if `x` is less than `y`.
+Values `x` and `y` are treated as integers unless `text` is set."""
 ))
-def func_lt(parser, x, y):
+def func_lt(parser, x, y, text=None):
```

Comment:
If we want to add more types of comparison, it would perhaps be better to have something like:

```python
def func_lt(parser, x, y, type=None):
    if type == 'text':
```

The idea behind this is that it will be possible to add new types, for example, 'date' or 'float'.
It will default to 'int', but this type should also be handled as a valid type.

Reply:
Good point.  I've modified all functions to allow specifying the processing type.  In addition, I added floating point processing to the `$gt`, `$gte`, `$lt` and `$lte` functions, and integer and floating point processing options to the `$max` and `$min` functions.